org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/filebased/RepositoryUtils.java

```diff
@@ -0,0 +1,201 @@
+/********************************************************************************
+ * Copyright (c) 2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache Software License 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *******************************************************************************/
+
+package org.eclipse.winery.repository.backend.filebased;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.LinkOption;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import org.eclipse.winery.common.RepositoryFileReference;
+import org.eclipse.winery.common.ids.GenericId;
+import org.eclipse.winery.common.ids.Namespace;
+import org.eclipse.winery.common.ids.admin.AdminId;
+import org.eclipse.winery.common.ids.definitions.DefinitionsChildId;
+import org.eclipse.winery.common.ids.elements.ToscaElementId;
+import org.eclipse.winery.repository.Constants;
+
+class RepositoryUtils {
+
+    private final MultiRepository multiRepository;
+
+    public RepositoryUtils(MultiRepository repository) {
+        this.multiRepository = repository;
+    }
+
+    protected void checkGitIgnore() throws IOException {
+        File ignore = new File(this.multiRepository.getRepositoryDep().toFile(), Constants.FILE_GIT_IGNORE);
+
+        if (!ignore.exists()) {
+            if (ignore.createNewFile()) {
+                BufferedWriter out = new BufferedWriter(new FileWriter(ignore));
+                out.write("# Will ignore any file except the repositories.json and the local contents of the workspace folder.");
+                out.newLine();
+                out.write("/**");
+                out.newLine();
+                out.write("!/" + Constants.DEFAULT_LOCAL_REPO_NAME + "/");
+                out.newLine();
+                out.write("!/" + Constants.DEFAULT_LOCAL_REPO_NAME + "/**");
+                out.close();
+            }
+            Files.setAttribute(ignore.toPath(), "dos:hidden", true, LinkOption.NOFOLLOW_LINKS);
```

Comment:
This will throw an `UnsupportedOperationException` on filesystems that do not support the `dos`  FileAttributeView